src/common/Requirements.ts

```diff
@@ -20,13 +20,15 @@ export interface Requirement {
     checkFunction: CheckRequirementsFunc;
     fulfilledMessage: string;
     unfulfilledMessage: string;
+    remediationMessage?: string;
```

Comment:
We may also want to name this something other than `remediationMessage`. Whether a step passed or failed, there may be extra details to be printed out for that step (even when it is passed).

So for example if we call this `detailedMessage` then when we're printing out results, regardless of passed/failed status we can also print out `detailedMessage` if it is not empty. Then in our product code whenever a stage check passes or fails, it would be up to our code to populate `detailedMessage` with appropriate content (if any) or just leave it blank.

Reply:
I like this too. I think there's a lot of potential for the success case messaging as well, primarily in the form of details of the satisfied requirement.  E.g. Xcode supported version fulfilled, and the details are "Running Xcode 12.2", things like that.

I'd propose we amend the existing flow here to allow for success and failure details, and update the naming convention accordingly.

Reply:
@maliroteh-sf @khawkins In the code I decided to call it a `supplementalMessage` instead as it is not a required message for a requirement. Please let me know if you rather like to have it `detailedMessage`.

Reply:
Sure that's fine (and probably a better name than `detailedMessage`) :)